### PR TITLE
feat: add cancel button for offline deals waiting for data

### PR DESCRIPTION
**Pull request by LexLuthr**
fixes https://github.com/filecoin-project/boost/issues/1403

Screenshot 2023-05-03 at 3 16 06 PM
Screenshot 2023-05-03 at 3 17 44 PM
Screenshot 2023-05-03 at 3 17 54 PM

### CODE REVIEW

**Comment by dirkmc** [comment]
Nice work 🙌 

I'd suggest that instead of creating a new mutation CancelOfflineDealAwaitingData we just use the existing dealCancel endpoint, and make it work for offline deals.
So in the DealCancel resolver, check if it's an offline deal, and if so call `provider.CancelOfflineDealAwaitingImport(dealUuid)`

**Comment by LexLuthr** [comment]
@dirkmc I have updated the mutations. I should have caught that during the initial commit.

**Review by dirkmc — APPROVED** [review]
One nit but otherwise LGTM 👍 

Please test manually to make sure it works after the refactor and then feel free to merge

File: react/src/DealDetail.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 import React, {useEffect, useState} from "react";
 import {useMutation, useQuery, useSubscription} from "@apollo/react-hooks";
-import {DealCancelMutation, DealFailPausedMutation, DealRetryPausedMutation, DealSubscription, EpochQuery} from "./gql";
+import {DealCancelMutation, DealFailPausedMutation, DealRetryPausedMutation, DealSubscription, EpochQuery, CancelOfflineDealAwaitingData} from "./gql";
```
[the remark this answers is not in the document]
Comment (LexLuthr) [reply react/src/DealDetail.js]: Tested again manually. Removed CancelOfflineDealAwaitingData from JS file.

Screenshot 2023-05-03 at 10 01 35 PM
Screenshot 2023-05-03 at 10 01 45 PM

**Pull request merged**